File: code/c/dynamic-memory/simple-node.c
/* struct example */
#include <stdlib.h>

typedef struct node_struct
{
    int                 value; // node value
    struct node_struct  *next; // pointer to next node
} node;

node *create_node(int val, node *next)
{
    node *result;
    result = (node *)malloc(sizeof(node));
    result->value = val;
    result->next = next;
    return result;
}

int main()
{
    node *current;
    current = create_node(0, NULL);
    current = create_node(1, current);
    return 0;
}

File: code/pascal/dynamic-memory/SimpleNode.pas
type 
    NodePtr = ^Node;
    Node = record
        value:  Integer; // node value
        next:   NodePtr; // pointer to next node
    end;

function CreateNode(val: Integer; next: NodePtr): NodePtr;
begin
    New(result);
    result^.value   := val;
    result^.next    := next;
end;

procedure Main();
var
    current: NodePtr;
begin
    current := CreateNode(0, nil);
    current := CreateNode(1, current);
end;

begin
    Main();
end.